### DISCUSSION TITLE
Not downloading the whole video

### DISCUSSION THREAD

**Discussion by TheRealBradA** [aclap-dev/video-downloadhelper#123] [Help and Q&A] [answered]
### Your question

Some of the videos aren't downloading fully (not all of them, some work fine). The process doesn't stop, it looks like finishes 100% but then when I go to play back there's only a portion of the whole video (54.39 out of 2.00.24 for example). I try and try again, thinking maybe it's my internet signal (stops when it drops out or falls below a certain threshold) but I don't see any reason that would be the case. And it's hard with not having premium so I can't just do it again, I have to wait. I would happily buy the premium if this is resolved and I could count on it fully., when it does work properly it works great. Here's a video link for example, I've tried this multiple times and it just won't play right https://vimeo.com/827342736
Note: this is in Chrome, but I've also used Firefox with the same issue. That browser is the most up to date version (version 122.0.1 [64-bit])

### Addon & CoApp version and details

version: 8.2.0.23
channel: stable
build date: 2024-02-14
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes

**Comment by jcfrog** [comment]
Hi, 
Thanks for the link but the video is private

**Reply by TheRealBradA** [reply]
password is [REDACTED]

**Comment by Mattmjs** [comment]
There is a problem on Vimeo videos if the duration is over 1 hour, in this instance your video is 2:00:46 long.
But VDH only says 0:46, it should reach 100% but it will continue downloading. Is this what you see ?
I've observed this on anything over 1 hour.

**Reply by paulrouget** [reply]
What's the url?

**Reply by Mattmjs** [reply]
Here one from the old google group: https://www.konserthuset.se/en/play/alfven-and-nilsson/

**Reply by paulrouget** [reply]
Thank you, I'll try to reproduce and investigate.

**Reply by paulrouget** [reply]
Right, so, the issue is that the website (internally) advertises a different duration than the actual video duration. So once the download reaches the advertised length, we see that there's more to it than expected, so we keep downloading, until there's nothing to download anymore.

Not much we can do about that. We don't know the full length.

Does it make sense?

**Reply by Mattmjs** [reply]
The concert video in the player is 1:49:36, in the VDH menu it will only see it as 49:36, it does that on any video over an hour.
So yeah it reaches 100% but continues until finished. But if I use ffmpeg that comes with the companion app and use the
master mpd link it will see the correct length. So that's a little puzzling

**Reply by paulrouget** [reply]
Yeah, the MPD advertise the wrong length, but ffmpeg is smart and after a HTTP request sees the proper length. I will update the duration after the first HTTP call.

You'll see that change in an upcoming update this weekend.

Should improve the situation.

**Reply by paulrouget** [reply]
@Mattmjs do you use Firefox or Chrome?

**Reply by Mattmjs** [reply]
I mainly use Firefox, I don't have chrome but I use an open source version called Ungoogled Chromium.

**Reply by paulrouget** [reply]
Ok - do you mind testing the beta? https://www.downloadhelper.net/w/Beta-versions

There's version 8.2.1.0a1 there. Which I believe will fix your issue.

**Reply by Mattmjs** [reply]
I tried the beta on the concert video and the menu still shows 49.36, but I see the download progressing with the % increasing until 
reaching 100. Then it stop with the complete file downloaded.

**Reply by paulrouget** [reply]
Awesome :)

**Comment by paulrouget** [comment]
> The process doesn't stop, it looks like finishes 100% but then when I go to play back there's only a portion of the whole video

I've explained what's going on here: https://github.com/aclap-dev/video-downloadhelper/discussions/123#discussioncomment-8498804

---

I hope to improve that situation in the future.

**Comment by paulrouget** [comment, marked as answer]
Ok, I have a working fix.

This will be part of an upcoming update this weekend.

**Reply by paulrouget** [reply]
In beta: https://github.com/aclap-dev/video-downloadhelper/discussions/142
